emoji-ui/src/components/emojiList.js

```diff
@@ -0,0 +1,27 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import EmojiItem from './emojiItem';
+import './emojiList.css';
+
+class EmojiList extends Component {
+  static propTypes = {
+    sourceEmoji: PropTypes.arrayOf(
+      PropTypes.shape({
+        isSelected: PropTypes.bool,
+        name: PropTypes.string,
+        url: PropTypes.string,
+      })
+    ),
+    clickHandler: PropTypes.func.isRequired,
+  };
+
+  render() {
+    const { sourceEmoji } = this.props;
+    const emojiListItem = sourceEmoji.map((emoji, idx) => {
```

Comment:
`key={idx} emoji={emoji}`  — much better is `key={emoji.name} emoji={emoji} `
the key is how react figures out which dom objects are identical for the purpose of minimizing dom-changes in re-rendering
so if you have a key that id’s an object you want to use that always
in this case, name identifies an emoji far more than its position in the list (esp. if we add search/filter)
so passing name as the key gives optimal diffing

Reply:
sounds good, will update!